README.md
# primtux-perte-mot-de-passe-ctparental

## Késako
Interface GUI pour la modification du couple Admin / Password de CTParental

Et c'est tout ;)


## Installer le script

```
wget https://github.com/CyrilleBiot/primtux-perte-mot-de-passe-ctparental/raw/main/primtux-password-ctparental-change_1.8_all.deb
sudo primtux-password-ctparental-change
```

Mais le mieux serait de le coller sur ce bouton :
![screenshoot](./screenshoot2.png)



## Screenshoot
![screenshoot](./screenshoot1.png)



debian/changelog
primtux-password-ctparental-change (1.8) UNRELEASED; urgency=medium

  * Fix bugs
  * Fix sym link
  * Fix typ
  * Fix python ref 
  * Add dialog box OK
  * Fix python dir path 
  * Hidden Fields password
  * Fix shell command

 -- Cyrille Biot <cyrille@cbiot.fr>  Thu, 04 Mar 2021 18:50:09 +0100

primtux-password-ctparental-change (1.0) unstable; urgency=medium

  * Initial release. (Closes: #XXXXXX)

 -- Cyrille Biot <cyrille@cbiot.fr>  Thu, 04 Mar 2021 14:25:48 +0100


perte-mot-de-passe-ctparental.sh
#/bin/bash
#
#__Source : https://github.com/CyrilleBiot/primtux-perte-mot-de-passe-ctparental 
#__author__ = "Cyrille BIOT <cyrille@cbiot.fr>"
#__copyright__ = "Copyleft"
#__site__ = "https://cbiot.fr"
#__license__ = "GPL"
#__version__ = "1.0"
#__email__ = "cyrille@cbiot.fr"
#__status__ = "Devel"
#__date__ = "2021/03/01"
#
#

# Test des droits d'execution
if [[ $EUID -ne 0 ]]; then
	echo "Ce script doit être lancé en root ou via sudo" 
	exit 1
fi

# Test si existence parametre IP 
if [[ $# -ne 2 ]]; then
	echo "ATTENTION. Ce script prend 2 paramètres."
	echo "Vérfier la syntaxe"
	echo "sudo bash perte-mot-de-passe-ctparental.sh <LOGIN> <PASSWORD>"
	exit 1
fi

if [[ -f  /var/www/CTadmin/arp1_md5.password  ]]; then
	echo "Le fichier de configuration existe, cool."
	else
	echo "Le fichier de configuration de CTparental ne semble pas exister."
	echo "Il doit y avoir un soucis plus profond. Contacter le forum primtux."
	echo "https://forum.primtux.fr/"
fi

# Nouveau fichier de configuration
  mv /var/www/CTadmin/arp1_md5.password /var/www/CTadmin/arp1_md5.password.BACKUP 
  MD5PWD=$(openssl passwd -apr1 -salt SALT $2)
  echo "$1:$MD5PWD" > /var/www/CTadmin/arp1_md5.password
  echo "Nouveau LOGIN : $1"
  echo "Nouveau MOT DE PASSE : $2"
  echo "Essayer de ne pas le perdre cette fois-ci..."


source/primtux-password-ctparental-change.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
  Source : https://github.com/CyrilleBiot/primtux-perte-mot-de-passe-ctparental
__author__ = "Cyrille BIOT <cyrille@cbiot.fr>"
__copyright__ = "Copyleft"
__credits__ = "Cyrille BIOT <cyrille@cbiot.fr>"
__license__ = "GPL"
__version__ = "1.0"
__date__ = "2021/03/03"
__maintainer__ = "Cyrille BIOT <cyrille@cbiot.fr>"
__email__ = "cyrille@cbiot.fr"
__status__ = "Devel"
"""

import os, subprocess
import gi
gi.require_version("Gtk", "3.0")
from gi.repository import Gtk, Gdk

class chgCtpPassword(Gtk.Window):
    def __init__(self):
        Gtk.Window.__init__(self, title="Primtux. Réinitialiser Compte Administration CTparental")

        if os.path.exists('.git'):
            self.pathDir = "./"
        # Launch since a deb package install
        else:
            self.pathDir = "/usr/share/primtux-password-ctparental-change/"

        self.set_border_width(10)
        self.set_resizable(False)
        self.set_icon_from_file(self.pathDir + "apropos.png")
        self.set_border_width(10)

        # Set Entry Widget to the login
        entryLogin = Gtk.Entry()
        entryLogin.set_text("Ici le nouveau login")
        entryLogin.set_width_chars(50)
        entryLogin.set_editable(True)

        # Set Entry Widget to the Password
        entryPwd1 = Gtk.Entry()
        entryPwd1.set_visibility(False)
        entryPwd1.set_width_chars(50)
        entryPwd1.set_editable(True)

        labelPwd1 = Gtk.Label(label="Mot de passe :")

        # Set Entry Widget to the Password
        entryPwd2 = Gtk.Entry()
        entryPwd2.set_visibility(False)
        entryPwd2.set_width_chars(50)
        entryPwd2.set_editable(True)

        labelPwd2 = Gtk.Label(label="Confirmer mot de passe :")


        # Set Button to Validate
        btnValidate = Gtk.Button(label="Générer Nouvelle association login / mot de passe")
        btnValidate.connect("clicked", self.testPassword, entryLogin, entryPwd1, entryPwd2)

        # Set label to rules for the new password
        lblInfo = Gtk.Label(label= "Le mot de passe doit contenir au moins\n"
                            "- 10 caractères \n"
                            "- des minuscules \n"
                            "- des majuscules \n"
                            "- des caractères spéciaux ($, @, #, %, (, ), [, ], !, :, ;)\n")
        lblInfo.set_line_wrap(True)
        lblInfo.set_justify(Gtk.Justification.LEFT)

        grid = Gtk.Grid()
        grid.set_column_spacing(6)
        grid.set_row_spacing(6)
        #grid.set_row_homogeneous(False)
        grid.attach(entryLogin, 0, 0, 1, 1)
        grid.attach(lblInfo,0,1,1,1)
        grid.attach(labelPwd1,0,2,1,1)
        grid.attach(entryPwd1, 1, 2, 1, 1)
        grid.attach(labelPwd2,0,3,1,1)
        grid.attach(entryPwd2, 1, 3, 1, 1)
        grid.attach(btnValidate,0,4,2,1)

        self.add(grid)

    def testPassword(self, button, login, password1, password2):
        """
        :param button:
        :param login: login
        :param password1: mot de passe 1
        :param password2: mot de passe 2
        :return:
        """

        # Récupération des mots de passe
        password1 = password1.get_text()
        password2 = password2.get_text()

        # Et du login
        login = login.get_text()

        # Cactères spéciaux
        SpecialSym = ['$', '@', '#', '%','(',')','[',']','!',':',';']

        # Chaine de test
        if password2 != password1:
            self.warning_alert(self,"Attention","Les 2 mots de passe ne sont pas identiques.")
        elif len(password1) < 10:
            self.warning_alert(self,"Attention","Le mot de passe doit contenir au moins 10 lettres.")
        elif not any(char.isdigit() for char in password1):
            self.warning_alert(self,"Attention","Le mot de passe doit contenir au moins 1 chiffre")
        elif not any(char.isupper() for char in password1):
            self.warning_alert(self,"Attention","Le mot de passe doit contenir au moins 1 lettre majuscule.")
        elif not any(char.islower() for char in password1):
            self.warning_alert(self,"Le mot de passe doit contenir au moins 1 lettre minuscule.")
        elif not any(char in SpecialSym for char in password1):
            self.warning_alert(self, "Attention", "Le mot de passe doit contenir au moins 1 caractère spcécial.")
        else:
            self.warning_OK(self,"Nickel", "Le mot de passe est conforme aux attentes. Cliquer sur OK/VALIDER.")
            # On lance la modification du mot de passe.
            cmd_shell = "sudo CTparental -setadmin " + login + " "  + password2 + " &> /dev/null"
            os.system(cmd_shell)
            self.warning_OK(self,"Nickel", "Le couple <LOGIN> / <MOT DE PASSE> a été mis à jour.")


    def gtk_style(self):
        style_provider = Gtk.CssProvider()
        style_provider.load_from_path(self.pathDir + 'style.css')

        Gtk.StyleContext.add_provider_for_screen(
            Gdk.Screen.get_default(),
            style_provider,
            Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION
        )
    def warning_alert(self, widget, message1, message2):
        dialog = Gtk.MessageDialog(
            transient_for=self,
            flags=0,
            message_type=Gtk.MessageType.ERROR,
            buttons=Gtk.ButtonsType.CANCEL,
            text=message1,
        )
        dialog.format_secondary_text(message2)
        dialog.run()
        dialog.destroy()

    def warning_OK(self, widget, message1, message2):
        """
        FOnction ouvrant une dialog box d'alerte
        :param widget:
        :param message1: Le titre du message
        :param message2: Le contenu du message
        :return:
        """
        dialog = Gtk.MessageDialog(
            transient_for=self,
            flags=0,
            message_type=Gtk.MessageType.INFO,
            buttons=Gtk.ButtonsType.OK,
            text=message1,
        )
        dialog.format_secondary_text(message2)
        dialog.run()
        dialog.destroy()




win = chgCtpPassword()
win.connect("destroy", Gtk.main_quit)
win.gtk_style()
win.show_all()
Gtk.main()